src/Lykke.Service.EthereumWorker.AzureRepositories/BlockIndexationLockRepository.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Threading.Tasks;
+using AzureStorage;
+using AzureStorage.Tables;
+using Common;
+using Lykke.Common.Log;
+using Lykke.Service.EthereumCommon.AzureRepositories;
+using Lykke.Service.EthereumWorker.AzureRepositories.Entities;
+using Lykke.Service.EthereumWorker.Core.Domain;
+using Lykke.Service.EthereumWorker.Core.Repositories;
+using Lykke.SettingsReader;
+
+
+namespace Lykke.Service.EthereumWorker.AzureRepositories
+{
+    public class BlockIndexationLockRepository : RepositoryBase, IBlockIndexationLockRepository
+    {
+        private readonly INoSQLTableStorage<BlockIndexationLockEntity> _locks;
+        
+        
+        private BlockIndexationLockRepository(
+            INoSQLTableStorage<BlockIndexationLockEntity> locks)
+        {
+            _locks = locks;
+        }
+        
+        
+        public static IBlockIndexationLockRepository Create(
+            IReloadingManager<string> connectionString,
+            ILogFactory logFactory)
+        {
+            var locks = AzureTableStorage<BlockIndexationLockEntity>.Create
+            (
+                connectionString,
+                "BlockIndexationLocks",
+                logFactory
+            );
+            
+            return new BlockIndexationLockRepository
+            (
+                locks: locks
+            );
+        }
+        
+        
+        public Task DeleteIfExistsAsync(
+            BigInteger blockNumber)
+        {
+            var (partitionKey, rowKey) = GetKeys(blockNumber);
+
+            return _locks.DeleteIfExistAsync
+            (
+                partitionKey: partitionKey,
+                rowKey: rowKey
+            );
+        }
+
+        public async Task<IEnumerable<BlockIndexationLock>> GetAsync()
+        {
+            return (await _locks.GetDataAsync())
```

Comment:
Full table scan. Are you sure that count of entities always will be reasonably small?

Reply:
Yes, I suppose, it always will be lower than 100